cinder/autogenerated_client.go

```diff
@@ -0,0 +1,1821 @@
+package cinder
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+type RequestHandlerFn func(*http.Request) (*http.Response, error)
+
+type UpdateVolumeTypeParams struct {
+
+	// VolumeType is required.
+	//
+	// A volume type offers a way to categorize or group volumes.
+	VolumeType string `json:"volume_type"`
+
+	//
+	// The unique identifier of the tenant or account.
+	TenantId string `json:"-"`
+
+	// VolumeTypeId is required.
+	//
+	// The unique identifier for an existing volume type.
+	VolumeTypeId string `json:"-"`
+}
+
+type UpdateVolumeTypeResults struct {
+	VolumeType struct {
+		ExtraSpecs struct {
+			Capabilities string `json:"capabilities"`
+		} `json:"extra_specs"`
+		ID   string `json:"id"`
+		Name string `json:"name"`
+	} `json:"volume_type"`
+}
+
+//
+// Updates a volume type.
+func updateVolumeType(request RequestHandlerFn, args UpdateVolumeTypeParams) (*UpdateVolumeTypeResults, error) {
```

Comment:
@wallyworld and I discussed this at one point. I think it's a good idea, and the only reason I can see to keep it in the same package is that we make extensive use of the autogenerated types. I'm not sure if it makes sense to have to reference a 2nd package for that. Any opinion on that?